bl/validation/main.go

```diff
@@ -5,6 +5,10 @@ import (
 	"strings"
 )
 
+type ValidationWarning struct {
```

Comment:
why use a struct with only 1 field, just do var ValidationWarning string

Reply:
easier and more intuitive to check for `nil` value if no warning message (which is optional)

Reply:
if `validationWarning.errorMesage != nil` is the same as checking if errorMessage != ""

but the addition of the struct adds more information to the code, no if I see it I may assume that there is something more to validationWarning other than the message and I have to go to the struct implementation to check it, whereas with a simple string I know immediately what it is

Reply:
`string` does not have a nil value. checking for `errorMessage != ""` is not the same as `!= nil`
I use a check for existing warning message in other places in the code and I think the `validationWarning` object should be treated same as `error` as far as checking for existance goes.

however, I think this discussion is taking too long so I will make the change to advance the pr